python_modules/libraries/dagster-airbyte/dagster_airbyte/components/workspace_component/component.py

```diff
@@ -0,0 +1,149 @@
+        },
+    )
+
+
+TranslationFn: TypeAlias = Callable[[dg.AssetSpec, AirbyteConnectionTableProps], dg.AssetSpec]
+ResolvedTranslationFn: TypeAlias = Annotated[
+    TranslationFn,
+    dg_components.Resolver(
+        resolve_translation,
+        model_field_type=Union[str, dg_components.AssetAttributesModel],  # type: ignore
+    ),
+]
+
+
+class ProxyDagsterAirbyteTranslator(DagsterAirbyteTranslator):
+    def __init__(self, fn: TranslationFn):
+        self.fn = fn
+
+    def get_asset_spec(self, props: AirbyteConnectionTableProps) -> dg.AssetSpec:
+        base_asset_spec = super().get_asset_spec(props)
+        spec = self.fn(base_asset_spec, props)
+
+        return spec
+
+
+class AirbyteCloudWorkspaceModel(BaseModel):
+    workspace_id: str = pydantic.Field(..., description="The Airbyte Cloud workspace ID.")
+    client_id: str = pydantic.Field(
+        ..., description="Client ID used to authenticate to Airbyte Cloud."
+    )
+    client_secret: str = pydantic.Field(
+        ..., description="Client secret used to authenticate to Airbyte Cloud."
+    )
+
+
+class AirbyteConnectionSelectorByName(BaseModel):
+    by_name: Sequence[str] = pydantic.Field(
+        ...,
+        description="A list of connection names to include in the collection.",
+    )
+
+
+class AirbyteConnectionSelectorById(BaseModel):
+    by_id: Sequence[str] = pydantic.Field(
+        ...,
+        description="A list of connection IDs to include in the collection.",
+    )
+
+
+def resolve_connection_selector(
+    context: dg_components.ResolutionContext, model
+) -> Optional[Callable[[AirbyteConnection], bool]]:
+    if isinstance(model, str):
+        model = context.resolve_value(model)
+
+    if isinstance(model, AirbyteConnectionSelectorByName):
+        return lambda connection: connection.name in model.by_name
+    elif isinstance(model, AirbyteConnectionSelectorById):
+        return lambda connection: connection.id in model.by_id
+    else:
+        check.failed(f"Unknown connection target type: {type(model)}")
+
+
+class AirbyteCloudWorkspaceComponent(Component, Model, Resolvable):
+    """Loads Airbyte Cloud connections from a given Airbyte Cloud workspace as Dagster assets.
+    Materializing these assets will trigger a sync of the Airbyte Cloud connection, enabling
+    you to schedule Airbyte Cloud syncs using Dagster.
+    """
+
+    workspace: Annotated[
+        AirbyteCloudWorkspace,
+        dg_components.Resolver(
+            lambda context, model: AirbyteCloudWorkspace(
+                **resolve_fields(model, AirbyteCloudWorkspace, context)  # type: ignore
+            )
+        ),
+    ]
+    connection_selector: Annotated[
+        Optional[Callable[[AirbyteConnection], bool]],
+        dg_components.Resolver(
+            resolve_connection_selector,
+            model_field_type=Union[
+                str, AirbyteConnectionSelectorByName, AirbyteConnectionSelectorById
+            ],
+        ),
```

Comment:
nit: maybe slap `description` on via `Resolver` since you did such a good job of adding it everywhere else